### PR TITLE
Fixing typos

### PR DESCRIPTION
**Pull request by deining**
This PR fixes a few typos and 2 broken links I spotted. I also improves other links and changes various links from `http` to `https` protocol.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -424,7 +424,7 @@ library from v2 to v3, it is hard to categorize changes in terms of sem
 
 ### Documentation
 
- - Update docs to use `prop-types` ([#894](https://github.com/enzymejs/enzyme/pull/894), [#890](https://github.com/enzymejs/enzyme/issue/890))
+ - Update docs to use `prop-types` ([#894](https://github.com/enzymejs/enzyme/pull/894), [#890](https://github.com/enzymejs/enzyme/issues/890))
```
Comment (ljharb) [inline CHANGELOG.md:427]: both of these links are correct, why the change?

File: README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ Enzyme
 
 [![Join the chat at https://gitter.im/enzymejs/enzyme](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/enzymejs/enzyme?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-[![npm Version](https://img.shields.io/npm/v/enzyme.svg)](https://www.npmjs.com/package/enzyme) [![License](https://img.shields.io/npm/l/enzyme.svg)](https://github.com/enzymejs/enzyme/blob/master/LICENSE.md) [![Build Status](https://travis-ci.org/enzymejs/enzyme.svg)](https://travis-ci.org/enzymejs/enzyme) [![Coverage Status](https://codecov.io/gh/enzymejs/enzyme/branch/master/graph/badge.svg)](https://codecov.io/gh/enzymejs/enzyme/branch/master)
+[![npm Version](https://img.shields.io/npm/v/enzyme.svg)](https://www.npmjs.com/package/enzyme) [![License](https://img.shields.io/npm/l/enzyme.svg)](https://github.com/enzymejs/enzyme/blob/master/LICENSE.md) [![Build Status](https://travis-ci.org/enzymejs/enzyme.svg)](https://travis-ci.org/enzymejs/enzyme) [![Coverage Status](https://codecov.io/gh/enzymejs/enzyme/branch/master/graph/badge.svg)](https://app.codecov.io/gh/enzymejs/enzyme/branch/master)
```
Comment (ljharb) [inline README.md:6]: it’s easier to keep using the shorter url

File: README.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ Running Enzyme Tests
 
 Enzyme is unopinionated regarding which test runner or assertion library you use, and should be
 compatible with all major test runners and assertion libraries out there. The documentation and
-examples for enzyme use [Mocha](https://mochajs.org/) and [Chai](http://chaijs.com/), but you
+examples for enzyme use [Mocha](https://mochajs.org) and [Chai](https://www.chaijs.com), but you
```
Comment (ljharb) [inline README.md:79]:
```suggestion
examples for enzyme use [Mocha](https://mochajs.org) and [Chai](https://chaijs.com), but you
```

File: CHANGELOG.md (hunk 1 above)
Comment (deining) [reply CHANGELOG.md:427]: > both of these links are correct, why the change?

No.

[https://github.com/enzymejs/enzyme/issue/890](https://github.com/enzymejs/enzyme/issue/890)

gives a `404 page not found` error for me.

File: README.md (hunk 2 above)
Comment (deining) [reply README.md:6]: > it’s easier to keep using the shorter url

Yes, **but**:

Go to:

https://enzymejs.github.io/enzyme/

and have a look at the badges at the start page:

```
coverage: unknown
```

This error will go away with this change. That's why I changed the URL.

File: README.md (hunk 3 above)
Comment (deining) [reply README.md]: I reverted this change via force-pushing the branch, as requested.

File: CHANGELOG.md (hunk 1 above)
Comment (ljharb) [reply CHANGELOG.md:427]: ah, that's right, it's only pulls/pull that works this way. thanks.

File: README.md (hunk 2 above)
Comment (ljharb) [reply README.md:6]: why would changing the destination URL have any impact on the image? I'd expect that to have to do with the image URL instead.

Comment (deining) [reply README.md]: > why would changing the destination URL have any impact on the image?

You are right. 

https://enzymejs.github.io/enzyme/ wasn't updated for a long time, that's the reason why the badge is not displaying correctly. I reverted the change to the destination URL The proper way to get rid of the broken badge is to redeploy the site on `github.io`.
